execution/block-partitioner/src/v2/mod.rs

```diff
@@ -0,0 +1,719 @@
+            DashMap::with_shard_amount(self.dashmap_num_shards);
+        let key_idx_table: DashMap<StateKey, StorageKeyIdx> =
+            DashMap::with_shard_amount(self.dashmap_num_shards);
+        let trackers: DashMap<StorageKeyIdx, RwLock<ConflictingTxnTracker>> =
+            DashMap::with_shard_amount(self.dashmap_num_shards);
+        for txn in txns.iter() {
+            senders.push(RwLock::new(None));
+            wsets.push(RwLock::new(HashSet::with_capacity(txn.write_hints().len())));
+            rsets.push(RwLock::new(HashSet::with_capacity(txn.read_hints().len())));
+        }
+        let timer_1 = MISC_TIMERS_SECONDS
+            .with_label_values(&["preprocess__main"])
+            .start_timer();
+        self.thread_pool.install(|| {
+            (0..num_txns)
+                .into_par_iter()
+                .for_each(|txn_idx: OriginalTxnIdx| {
+                    let txn = &txns[txn_idx];
+                    let sender = txn.sender();
+                    let sender_idx = *sender_idx_table
+                        .entry(sender)
+                        .or_insert_with(|| num_senders.fetch_add(1, Ordering::SeqCst));
+                    *senders[txn_idx].write().unwrap() = Some(sender_idx);
+                    let num_writes = txn.write_hints().len();
+                    for (i, storage_location) in txn
+                        .write_hints()
+                        .iter()
+                        .chain(txn.read_hints().iter())
+                        .enumerate()
+                    {
+                        let key = storage_location.state_key().clone();
+                        let key_idx = *key_idx_table
+                            .entry(key)
+                            .or_insert_with(|| num_keys.fetch_add(1, Ordering::SeqCst));
+                        let is_write = i < num_writes;
+                        if is_write {
+                            wsets[txn_idx].write().unwrap().insert(key_idx);
+                        } else {
+                            rsets[txn_idx].write().unwrap().insert(key_idx);
+                        }
+                        trackers
+                            .entry(key_idx)
+                            .or_insert_with(|| {
+                                let anchor_shard_id =
+                                    get_anchor_shard_id(storage_location, num_executor_shards);
+                                RwLock::new(ConflictingTxnTracker::new(
+                                    storage_location.clone(),
+                                    anchor_shard_id,
+                                ))
+                            })
+                            .write()
+                            .unwrap()
+                            .add_candidate(txn_idx, is_write);
+                    }
+                });
+        });
+        let duration_1 = timer_1.stop_and_record();
+        info!("preprocess__main={duration_1}");
```

Comment:
Let's remove them if they are temporarily added or convert them to `trace` if we plan to keep them. I would prefer removing them as we already have counters for them?